tests/common-platform-and-delius/search-for-person.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { DateTime } from 'luxon'
+import { hearingData } from '../../steps/court-case/hearing-data'
+import { buildAddress } from '../../steps/delius/address/create-address'
+
+dotenv.config() // read environment variables into process.env
+
+test('Create and search for a person', async ({ page }) => {
+    const person = deliusPerson()
+    const address = buildAddress()
+
+    await runPod(
+        'hmpps-probation-integration-services-dev',
+        'probation-integration-e2e-test',
+        'common-platform-and-delius',
+        ['aws sqs send-message --queue-url "$QUEUEURL" --message-body "$MESSAGEBODY" || sleep 30'],
+        [
+            {
+                name: 'QUEUEURL',
+                value: 'https://eu-west-2.queue.amazonaws.com/754256621582/probation-integration-dev-common-platform-and-delius-queue',
+            },
+            {
+                name: 'MESSAGEBODY',
+                value: JSON.stringify({
+                    Message: JSON.stringify(hearingData(person, address)),
+                }),
+            },
+        ]
+    )
+
+    await deliusLogin(page)
+    // Search for created person record via national search screen
+    await page.getByRole('link', { name: 'National Search' }).click()
+    await page.getByLabel('First Name or Preferred Name:').fill(person.firstName)
+    await page.getByLabel('Last Name:').fill(person.lastName)
+    await page.getByRole('button', { name: 'Search', exact: true }).click()
+    await expect(page.getByRole('cell', { name: person.lastName + ', ' + person.firstName, exact: true })).toBeVisible()
+    await page.getByRole('link', { name: 'View' }).nth(2).click()
+
+    // Check personal details screen information is correct
+    await page.getByRole('link', { name: 'Personal Details' }).click()
+    await expect(page.locator('[id="FirstName\\:outputText"]')).toContainText(person.firstName)
+    await expect(page.locator('[id="Surname\\:outputText"]')).toContainText(person.lastName)
+    await expect(page.locator('[id="Sex\\:outputText"]')).toContainText(person.sex)
+    await expect(page.locator('[id="DateOfBirth\\:DateOfBirth\\:outputText"]')).toContainText(
+        DateTime.fromJSDate(person.dob).toFormat('dd/MM/yyyy')
+    )
+    await expect(page.locator('[id="PNC\\:outputText"]')).toContainText(person.pnc)
+
+    // Check address record has been created
+    await page.getByRole('link', { name: 'Addresses' }).click()
+    await page.getByRole('link', { name: 'view', exact: true }).click()
+    await expect(page.locator('[id="j_idt714\\:outputText"]')).toContainText(address.streetAddress)
+    await expect(page.locator('[id="j_idt718\\:outputText"]')).toContainText(address.cityName)
+    await expect(page.locator('[id="j_idt722\\:outputText"]')).toContainText(address.county)
+    await expect(page.locator('[id="j_idt726\\:outputText"]')).toContainText(address.country)
+    await expect(page.locator('[id="j_idt730\\:outputText"]')).toContainText(address.zipCode)
```

Comment:
It would be good to avoid relying on ids if we can help it.  Playwright has additional locators which are handy for things like this.  For example, you could use the [layout-based locators](https://playwright.dev/docs/other-locators#css-matching-elements-based-on-layout) to find the element "next to" the label like we do here: https://github.com/ministryofjustice/hmpps-probation-integration-e2e-tests/blob/a7dace13de85242e24c44bd6c1d2254114d8ac38/tests/opd-and-delius/opd-assessment.spec.ts#L44

Reply:
I've modified test to use layout-based locators instead of id locators as suggested. Thanks!